mollie-payments-for-woocommerce/includes/mollie/wc/gateway/abstractseparecurring.php

```diff
@@ -0,0 +1,181 @@
+            self::WAITING_CONFIRMATION_PERIOD_DAYS
+        );
+
+        $paymentMethodTitle = $this->getPaymentMethodTitle($payment);
+
+        $renewal_order->add_order_note(sprintf(
+        /* translators: Placeholder 1: Payment method title, placeholder 2: payment ID */
+            __('%s payment started (%s).', 'mollie-payments-for-woocommerce'),
+            $paymentMethodTitle,
+            $payment->id . ($payment->mode == 'test' ? (' - ' . __('test mode', 'mollie-payments-for-woocommerce')) : '')
+        ));
+
+        $this->addPendingPaymentOrder($renewal_order);
+    }
+
+    /**
+     * @param $renewal_order
+     */
+    protected function addPendingPaymentOrder($renewal_order)
+    {
+        global $wpdb;
+
+        $confirmationDate = new DateTime();
+        $period = 'P'.self::WAITING_CONFIRMATION_PERIOD_DAYS . 'D';
+        $confirmationDate->add(new DateInterval($period));
+
+        $wpdb->insert(
+            $wpdb->mollie_pending_payment,
+            array(
+                'post_id' => $renewal_order->id,
+                'expired_time' => $confirmationDate->getTimestamp(),
+            )
+        );
+    }
+
+    /**
+     * @param $order
+     */
+    protected function deleteOrderFromPendingPaymentQueue($order)
+    {
+        global $wpdb;
+        $wpdb->delete(
+            $wpdb->mollie_pending_payment,
+            array(
+                'post_id' => $order->id,
+            )
+        );
+    }
+
+    /**
+     * @param WC_Order $order
+     * @param Mollie_API_Object_Payment $payment
+     */
+    protected function onWebhookPaid(WC_Order $order, Mollie_API_Object_Payment $payment)
+    {
+        parent::onWebhookPaid($order, $payment);
+        $this->deleteOrderFromPendingPaymentQueue($order);
+    }
+
+    /**
+     * @param WC_Order $order
+     * @param Mollie_API_Object_Payment $payment
+     */
+    protected function onWebhookCancelled(WC_Order $order, Mollie_API_Object_Payment $payment)
+    {
+        parent::onWebhookCancelled($order, $payment);
+        $this->deleteOrderFromPendingPaymentQueue($order);
+    }
+
+    /**
+     * @param WC_Order $order
+     * @param Mollie_API_Object_Payment $payment
+     */
+    protected function onWebhookExpired(WC_Order $order, Mollie_API_Object_Payment $payment)
+    {
+        parent::onWebhookExpired($order, $payment);
+        $this->deleteOrderFromPendingPaymentQueue($order);
+    }
+
+    /**
+     * @param null $payment
+     * @return string
+     */
+    protected function getPaymentMethodTitle($payment)
+    {
+        $paymentMethodTitle = parent::getPaymentMethodTitle($payment);
+        if ($payment->method == $this->getRecurringMollieMethodId()){
+            $paymentMethodTitle = $this->getRecurringMollieMethodTitle();
+        }
+
+        return $paymentMethodTitle;
+    }
+
+    /**
+     * @param $order
+     * @param $payment
+     */
+    protected function handlePayedOrderWebhook($order, $payment)
```

Comment:
Volgens mij hoeft deze methode niet overschreven te worden. Als ik het goed begrijp wordt deze methode alleen aangeroepen in `Mollie_WC_Gateway_Abstract::webhookAction()` als dit de tweede keer is dat een webhook wordt aangeroepen en deze webhook dus genegeerd kan worden.

Hij kijkt naar `orderNeedsPayment()` om te bepalen of hij een webhook verwacht. Ik kan zelf niet inschatten of die ook al rekening houdt met recurring payments.